### PR TITLE
Iris refactor

### PR DESCRIPTION
**Pull request by Istar-Eldritch**
This PR:

- Substitutes vault calls with IRIS
- Provides the trending datasets through IRIS with the pattern `action.dataset.trending`
- Implements the computation on intervals in a save way.

### CODE REVIEW

**Comment by DrWilliamsSteven** [comment]
@Istar-Eldritch I'll try to do this this afternoon..

File: src/trending-datasets.spec.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    const requestResult = [Math.random(), Math.random(), Math.random(), Math.random()];
+    const shuffledResult = [Math.random(), Math.random(), Math.random(), Math.random()];
+    const _shuffleArray = stub().returns(shuffledResult);
+    const request = stub().returns(Promise.resolve(requestResult));
+
+    const result = await query({_shuffleArray, request});
+
+    t.ok(request.calledOnce, 'It calls requests with the query');
+    const sCall = _shuffleArray.getCall(0);
+    t.deepEquals(sCall.args[0], requestResult, 'shuffles the response of the request');
```
Comment (DrWilliamsSteven) [inline src/trending-datasets.spec.ts]: How does this test work?! 
I'm missing something here.. 
Does `sCall.args[0]` return `shuffledResult`? How can `shuffledResult` equal `requestResult`? 
Or am I confusing something? :/

Comment (Istar-Eldritch) [inline src/trending-datasets.spec.ts]: You are right, there is something fishy here. I'll take a look :+1:

Comment (Istar-Eldritch) [inline src/trending-datasets.spec.ts]: You are right, there is something fishy here.

**Comment by Istar-Eldritch** [comment]
I just added more into to the README.md to clarify how to run this service in isolation without the rest of the platform ;) You need to run a standalone rabbit instance

File: src/trending-datasets.spec.ts (hunk 1 above)
Comment (Istar-Eldritch) [inline src/trending-datasets.spec.ts]: Took a second look. It's ok. what this checks is that shuffle is being called with the results of the request.

Comment (DrWilliamsSteven) [inline src/trending-datasets.spec.ts]: Ah yeah ok that makes sense. Got it!

**Comment by DrWilliamsSteven** [comment]
@Istar-Eldritch 
I wasn't able to get this to run locally yet..
Do you want me to approve, or wait til tomorrow?

**Comment by Istar-Eldritch** [comment]
It can wait :smile:

**Pull request merged**